src/resolvers/api/getNode.js

```diff
@@ -0,0 +1,147 @@
+const _get = require('lodash.get')
+
+const { query } = require('../utils')
+
+const getNode = async (_, args) => {
+  const { urbitId } = _get(args, 'input.urbitId') || null
+
+  let nodeType = null
+
+  if (urbitId.length === 4) {
+    nodeType = 'GALAXY'
+  } else if (urbitId.length === 7) {
+    nodeType = 'STAR'
+  } else if (urbitId.length === 14) {
+    nodeType = 'PLANET'
+  }
+
+  const getNumOwnersQuery = `select count(*) from pki_events where node_id = '${urbitId}' and event_type_id = '1';`
+  const getNumOwnersResponse = await query(getNumOwnersQuery)
+
+  const numOwners = parseInt(_get(getNumOwnersResponse, 'rows[0].count')) || 1
+
+  const sponsors = []
+
+  // sponsor
+  const getSponsorQuery = `select sponsor_id from pki_events where node_id = '${urbitId}' order by time desc limit 1;`
+  const getSponsorResponse = await query(getSponsorQuery)
+  const getSponsorResponseRow = _get(getSponsorResponse, 'rows[0]') || []
+
+  const sponsor = getSponsorResponseRow ? _get(getSponsorResponseRow, 'sponsor_id') || null : null
+
+  sponsors.push(sponsor)
+
+  // sponsor's sponsor
+  const getSponsorsSponsorQuery = `select sponsor_id from pki_events where node_id = '${sponsor}' order by time desc limit 1;`
+  const getSponsorsSponsorResponse = await query(getSponsorsSponsorQuery)
+  const getSponsorsSponsorResponseRow = _get(getSponsorsSponsorResponse, 'rows[0]') || []
+  const sponsorsSponsor = getSponsorsSponsorResponseRow ? _get(getSponsorsSponsorResponseRow, 'sponsor_id') || null : null
+
+  if (sponsorsSponsor) {
+    sponsors.push(sponsorsSponsor)
+  }
```

Comment:
A node only has one sponsor. If I was using this API from a client and I wanted to show my sponsor's sponsor, I'd expect to use the API roughly as follows:

```graphql
getNode(input: { urbitId: "~wolref-podlex" }) {
  sponsor {
    urbitId
    sponsor {
      urbitId
    }
  }
}
```

Reply:
In the [bounty description](https://grants.urbit.org/bounties/1101665172), `get-node` returns `sponsors: [node] (sponsor and sponsor's sponsor, if applicable)`. Does this contradict that? Maybe I'm misunderstanding something.

Reply:
Since based on discussion in another comment it's OK to leave it as an array of sponsors, can we leave this particular item as-is?

Reply:
After having spent some time working on the materialized view approach for nodes, I think not having having an array of sponsors is indeed the desirable approach here.